### ISSUE TITLE
dartdoc fails for package submitted to pub.dev but generates docs successfully when run locally. 

### ISSUE THREAD

**Issue by simphotonics** [dart-lang/dartdoc#2420]
The online documentation for package [`sample_statistics`](https://pub.dev/packages/sample_statistics) could not be generated.  
The error message posted under the *Scores* tab is:
```Emacs
Unhandled exception:
NoSuchMethodError: The getter 'documentationComment' was called on null.
Receiver: null
Tried calling: documentationComment
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:51:5)
#1      Accessor.computeDocumentationComment (package:dartdoc/src/model/accessor.dart:67:44)
#2      DocumentationComment.documentationComment (package:dartdoc/src/model/documentation_comment.dart:39:31)
#3      ModelElement.computeDocumentationFrom (package:dartdoc/src/model/model_element.dart:589:9)
#4      ModelElement.documentationFrom (package:dartdoc/src/model/model_element.dart:569:28)
#5      PackageGraph.precacheLocalDocs.<anonymous closure>.precacheOneElement.<anonymous closure> (package:dartdoc/src/model/package_graph.dart:109:16)
#6      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:165:25)
#7      Future.wait (dart:async/future.dart:406:26)
#8      PackageGraph.initializePackageGraph (package:dartdoc/src/model/package_graph.dart:74:18)
#9      PubPackageBuilder.buildPackageGraph (package:dartdoc/src/model/package_builder.dart:77:20)
<asynchronous suspension>
#10     Dartdoc.generateDocsBase (package:dartdoc/dartdoc.dart:182:41)
#11     Dartdoc.generateDocs (package:dartdoc/dartdoc.dart:223:34)
#12     main (file:///project/pkg/pub_dartdoc/bin/pub_dartdoc.dart:32:33)
<asynchronous suspension>
#13     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#14     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)
```

On my local machine `dartdoc` successfully generates the documentation.
```Console
$ dart --version
Dart SDK version: 2.10.3 (stable) (Unknown timestamp) on "linux_x64"

$ dartdoc --version
dartdoc version: 0.32.4

$ dartdoc
Documenting sample_statistics...
Initialized dartdoc with 59 libraries in 19.5 seconds
Generating docs for library sample_statistics from package:sample_statistics/sample_statistics.dart...
Validating docs...
no issues found
Documented 1 public library in 1.8 seconds
Success! Docs generated into /home/dan/sample_statistics/doc/api
```

**Comment by CaptainIRS** [comment]
This is the same issue as that of #2402. It was fixed in #2404, but pub.dev hasn't yet upgraded to a version of dartdoc with the fix. Once dart-lang/pub-dev#4218 lands, it should be fixed in pub.dev.

**Comment by simphotonics** [comment]
@CaptainIRS, thank you for your reply.

**Comment by jcollins-g** [comment]
duplicate of #2402
